metals/src/main/scala/scala/meta/internal/metals/codeactions/CreateCompanionObjectCodeAction.scala

```diff
@@ -0,0 +1,181 @@
+package scala.meta.internal.metals.codeactions
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+
+import scala.meta.Defn
+import scala.meta.Template
+import scala.meta.Tree
+import scala.meta.inputs.Position
+import scala.meta.internal.metals.Buffers
+import scala.meta.internal.metals.CodeAction
+import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.internal.metals.ServerCommands
+import scala.meta.internal.parsing.Trees
+import scala.meta.io.AbsolutePath
+import scala.meta.pc.CancelToken
+
+import org.eclipse.lsp4j.CodeActionParams
+import org.eclipse.lsp4j.Location
+import org.eclipse.{lsp4j => l}
+
+/**
+ * It creates braceless or braceful companion objects for classes, traits, and enums
+ * Then navigates to the position of the created object!
+ *
+ * @param trees
+ */
+class CreateCompanionObjectCodeAction(
+    trees: Trees,
+    buffers: Buffers
+) extends CodeAction {
+  override def kind: String = l.CodeActionKind.RefactorRewrite
+
+  override def contribute(params: CodeActionParams, token: CancelToken)(implicit
+      ec: ExecutionContext
+  ): Future[Seq[l.CodeAction]] = Future {
+    val uri = params.getTextDocument().getUri()
+
+    val path = uri.toAbsolutePath
+    val range = params.getRange()
+    val applyTree =
+      if (range.getStart == range.getEnd)
+        trees
+          .findLastEnclosingAt[Tree](
+            path,
+            range.getStart(),
+            applyWithSingleFunction
+          )
+      else
+        None
+
+    val maybeCompanionObject = for {
+      tree <- applyTree
+      name <- getNameOfClassTraitOrEnumTree(tree)
+      if !hasCompanionObject(tree, name)
+      document <- buffers.get(path)
+    } yield buildCreatingCompanionObjectCodeAction(
+      path,
+      tree,
+      uri,
+      getIndentationForPositionInDocument(tree.pos, document),
+      name,
+      hasBraces(tree, document)
+    )
+
+    maybeCompanionObject.toSeq
+
+  }
+
+  private def getIndentationForPositionInDocument(
+      treePos: Position,
+      document: String
+  ): String =
+    document
+      .substring(treePos.start - treePos.endColumn + 1, treePos.start)
```

Comment:
This seems to work for me:
```suggestion
      .substring(treePos.start - treePos.startColumn, treePos.start)
```